File: ARCADE/Intro/Level 2_4: almostlncreasingSequece.cs
    using System;
     
    public class Test
    {
    	public static bool almostIncreasingSequence(int[] sequence){
            int count = 0;
            for (int i = 0; i < sequence.Length - 1; i++){
                if (sequence[i] >= sequence[i + 1]){
                    count += 1;
                    if ((i - 1 >= 0) && (sequence[i - 1] >= sequence[i + 1])){
                        if ((sequence.Length - 2 > i) && (sequence[i] >= sequence[i + 2])) return false;
                    }
                }
            }
            return count <= 1 ? true : false;
        }
    	public static void Main()
    	{
    		int [] sequence = new int []{3, 1, 2};
    		Console.WriteLine(almostIncreasingSequence(sequence));
    	}
    }

File: ARCADE/Intro/Level 2_4: almostlncreasingSequece.pas
    program almostIncreasingSequence;
    Type ArrayInteger = Array of integer;
     
    function almostIncreasingSequence(sequence: ArrayInteger): boolean;
    var count, i : integer;
    begin
        count := 0;
        for i := 0 to length(sequence) - 2 do
        begin
            if(sequence[i] >= sequence[i+1]) then
            begin
                count := count + 1;
                if((i - 1 >= 0) and (sequence[i-1] >= sequence[i+1])) then
                    if((length(sequence) - 2 > i) and (sequence[i] >= sequence[i+2])) then 
                    begin 
                        almostIncreasingSequence := false;
                        count := 3;
                    end;
            end;
        end;
        if(count <= 1) then almostIncreasingSequence := true
        else almostIncreasingSequence := false;
    end;
     
    var sequence : ArrayInteger;
    begin
    	sequence := ArrayInteger.Create(1,2,1,2);
    	writeln(almostIncreasingSequence(sequence));
     
    	sequence := ArrayInteger.Create(1,3,2);
    	writeln(almostIncreasingSequence(sequence));
    end.
